core/jvm/src/main/scala/fs2/compression/CompressionPlatform.scala

```diff
@@ -262,747 +234,141 @@ private[compression] trait CompressionCompanionPlatform {
-      private object gzipCompressionMethod {
-        val DEFLATE: Byte = Deflater.DEFLATED.toByte
-      }
-      private object gzipFlag {
-        def apply(flags: Byte, flag: Byte): Boolean = (flags & flag) == flag
-        def apply(flags: Byte, flag: Int): Boolean = (flags & flag) == flag
-
-        def ftext(flags: Byte): Boolean = apply(flags, FTEXT)
-        def fhcrc(flags: Byte): Boolean = apply(flags, FHCRC)
-        def fextra(flags: Byte): Boolean = apply(flags, FEXTRA)
-        def fname(flags: Byte): Boolean = apply(flags, FNAME)
-        def fcomment(flags: Byte): Boolean = apply(flags, FCOMMENT)
-        def reserved5(flags: Byte): Boolean = apply(flags, RESERVED_BIT_5)
-        def reserved6(flags: Byte): Boolean = apply(flags, RESERVED_BIT_6)
-        def reserved7(flags: Byte): Boolean = apply(flags, RESERVED_BIT_7)
-
-        val FTEXT: Byte = 1
-        val FHCRC: Byte = 2
-        val FEXTRA: Byte = 4
-        val FNAME: Byte = 8
-        val FCOMMENT: Byte = 16
-        val RESERVED_BIT_5 = 32
-        val RESERVED_BIT_6 = 64
-        val RESERVED_BIT_7: Int = 128
-      }
-      private object gzipExtraFlag {
-        val DEFLATE_MAX_COMPRESSION_SLOWEST_ALGO: Byte = 2
-        val DEFLATE_FASTEST_ALGO: Byte = 4
-      }
-      private val gzipOptionalExtraFieldLengthBytes = 2
-      private val gzipHeaderCrcBytes = 2
-      private object gzipOperatingSystem {
-        val FAT_FILESYSTEM: Byte = 0
-        val AMIGA: Byte = 1
-        val VMS: Byte = 2
-        val UNIX: Byte = 3
-        val VM_CMS: Byte = 4
-        val ATARI_TOS: Byte = 5
-        val HPFS_FILESYSTEM: Byte = 6
-        val MACINTOSH: Byte = 7
-        val Z_SYSTEM: Byte = 8
-        val CP_M: Byte = 9
-        val TOPS_20: Byte = 10
-        val NTFS_FILESYSTEM: Byte = 11
-        val QDOS: Byte = 12
-        val ACORN_RISCOS: Byte = 13
-        val UNKNOWN: Byte = 255.toByte
-
-        val THIS: Byte = System.getProperty("os.name").toLowerCase() match {
-          case name if name.indexOf("nux") > 0  => UNIX
-          case name if name.indexOf("nix") > 0  => UNIX
-          case name if name.indexOf("aix") >= 0 => UNIX
-          case name if name.indexOf("win") >= 0 => NTFS_FILESYSTEM
-          case name if name.indexOf("mac") >= 0 => MACINTOSH
-          case _                                => UNKNOWN
-        }
-      }
-      private val gzipInputCrcBytes = 4
-      private val gzipInputSizeBytes = 4
-      private val gzipTrailerBytes = gzipInputCrcBytes + gzipInputSizeBytes
-
-      private val zeroByte: Byte = 0
-
-      private val fileNameBytesSoftLimit =
-        1024 // A limit is good practice. Actual limit will be max(chunk.size, soft limit). Typical maximum file size is 255 characters.
-      private val fileCommentBytesSoftLimit =
-        1024 * 1024 // A limit is good practice. Actual limit will be max(chunk.size, soft limit). 1 MiB feels reasonable for a comment.
+//              .onFinalize {
```

Comment:
* [ ] figure this out or remove the commented out piece

Reply:
https://discord.com/channels/632277896739946517/632310980449402880/927241378415640716

Reply:
As @mpilquist explained here: https://github.com/typelevel/fs2/issues/2785#issuecomment-1008046506 – we can't rely on `Stream.bracket` here. Changed it to use `Pull.bracketCase`.